app/platform_stats/rest.py

```diff
@@ -75,17 +75,18 @@ def get_data_for_billing_report():
     ]
     combined = {}
     for s in sms_costs:
-        entry = {
-            "organisation_id": str(s.organisation_id) if s.organisation_id else "",
-            "organisation_name": s.organisation_name or "",
-            "service_id": str(s.service_id),
-            "service_name": s.service_name,
-            "sms_cost": float(s.sms_cost),
-            "sms_fragments": s.chargeable_billable_sms,
-            "letter_cost": 0,
-            "letter_breakdown": ""
-        }
-        combined[s.service_id] = entry
+        if float(s.sms_cost) > 0:
```

Comment:
I think this is new since my original review. Could we DRY it up to avoid setting things like `organisation_id` in two places?

One way to do this would be to define a function to return the common `dict` between the two loops e.g. `initial_report_row_for_service`.

Another way would be to rewrite the three loops by doing the report in two steps:

1. "aggregation". First loop over the three arrays to generate a dict of the form `service_id => { "sms_cost" => ..., "letter_cost" => ..., "billing_details" => ...}`

2. "generation". Loop of the dict to generate the row for each service, depending on the keys in its dict (i.e. whether it has a `letter_cost` and/or `sms_cost`).

What do you think?

Reply:
I just added a new conditional (`if float(s.sms_cost`). The rest is not added by me, it just has different indentation. If you feel like refactoring this bit of code, feel free to add a commit for it to this PR, and I could review that commit then? :)

Reply:
Oh yes, I see now. That's fair to leave it as-is.

I wouldn't want to add a commit to this PR, as that would make the review situation ambiguous. I'd be up for it as a separate PR if I were making the original change. I don't feel strongly enough about it to pick it up myself.